src/inmanta/db/versions/v202411140.py

```diff
@@ -0,0 +1,100 @@
+"""
+    Copyright 2024 Inmanta
+
+    Licensed under the Apache License, Version 2.0 (the "License");
+    you may not use this file except in compliance with the License.
+    You may obtain a copy of the License at
+
+        http://www.apache.org/licenses/LICENSE-2.0
+
+    Unless required by applicable law or agreed to in writing, software
+    distributed under the License is distributed on an "AS IS" BASIS,
+    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+    See the License for the specific language governing permissions and
+    limitations under the License.
+
+    Contact: code@inmanta.com
+"""
+
+from asyncpg import Connection
+
+
+async def update(connection: Connection) -> None:
+    """
+    Add `resource_status` field to the resource_persistent_state table and populate it.
+    """
+    schema = """
+        CREATE TYPE public.new_resource_status AS ENUM ('UP_TO_DATE', 'HAS_UPDATE', 'UNDEFINED', 'ORPHAN');
```

Comment:
I don't know if enum's in the database are super usefull? I.e. I think they are kind of messy to migrate and we already have the type checks?

(It may have advantages as well, I don't know)

Reply:
I am also not a big fan of them. I remember I once said that we shouldn't use them anymore, but @sanderr disagreed to that. I don't recall the motivation. @sanderr Could you give your input on this.

Reply:
> I don't recall the motivation.

Me neither, so I'll give some fresh input. ~~One potential risk I see is with evolving code: if we use enums in Python and ints in the database, we must make sure their int representations remain stable. We can mitigate that risk by not using `enum.auto()` and adding a comment to the enum definitions that the values must remain stable.~~ EDIT: We probably can't do ints.

Iirc the recommended (by Markus) approach, was to use (constrained?) ints, and then to have another table mapping ints to names (essentially defining the enum). I'm not sure how feasible that change would be, given that our data access framework is already in place. I think that this was one of my motivations for sticking with the enum: tough to migrate to the right approach, so might as well stick to the current approach instead of moving to another "not best-practice" one.

Other than that, and the minor inconvenience when working with the database directly, I don't have any other objections. And I do agree there are benefits as well.

Overall, if we can easily move to the best-practice approach, and we can integrate it cleanly, I think I'm in favor. If not, do I understand correctly you'd go for `varchar[]?` in the database? I'm more on the fence for that one, but I'm willing to give it a shot and see how we like it down the line.